Index: mds-0.1/build/build.mk
===================================================================
--- mds-0.1.orig/build/build.mk
+++ mds-0.1/build/build.mk
@@ -137,7 +137,7 @@ $(project_name)_projects_used ?= $(proje
 # config-specific makefile.
 
 
-install_dir ?= $(git_base_dir)/install
+install_dir  = ${DESTDIR}
 $(project_name)_install_dir ?= $(install_dir)
 
 install_includes_dir ?= $($(project_name)_install_dir)/include
@@ -220,7 +220,7 @@ $(project_name)_debug_flag ?= $(debug_fl
 extra_cxx_flags ?= -Wall -fmessage-length=0 -pthread -fPIC
 $(project_name)_extra_cxx_flags ?= $(extra_cxx_flags)
 
-CPPFLAGS ?= \
+CPPFLAGS += \
 	$(std_flag) \
 	$($(project_name)_cpp_defines) \
 	$($(project_name)_cpp_includes) \
@@ -228,16 +228,16 @@ CPPFLAGS ?= \
 	$($(project_name)_extra_cpp_flags)
 $(project_name)_CPPFLAGS ?= $(CPPFLAGS)
 
-CXXFLAGS ?= \
+CXXFLAGS += \
 	$($(project_name)_opt_flag) \
 	$($(project_name)_debug_flag) \
 	$($(project_name)_extra_cxx_flags)
 $(project_name)_CXXFLAGS ?= $(CXXFLAGS)
 
-LDFLAGS ?=
+LDFLAGS +=
 $(project_name)_LDFLAGS ?= $(LDFLAGS)
 
-LIBS ?= -lstdc++ -lpthread
+LIBS += -lstdc++ -lpthread
 $(project_name)_LIBS ?= $(LIBS)
 
 ######################################
Index: mds-0.1/build/project.mk
===================================================================
--- mds-0.1.orig/build/project.mk
+++ mds-0.1/build/project.mk
@@ -28,8 +28,6 @@ include ../defs.mk
 
 project_name = mds_core
 
-projects_used = mpgc ruts
-
 mpgc_project_dir ?= $(if $(wildcard $(git_base_dir)/mpgc/build),$(git_base_dir)/mpgc,$(git_base_dir)/gc)
 ruts_project_dir ?= $(if $(wildcard $(mpgc_project_dir)/ruts/build),$(mpgc_project_dir)/ruts,$(git_base_dir)/common)
 
Index: mds-0.1/java-api/build/build.mk
===================================================================
--- mds-0.1.orig/java-api/build/build.mk
+++ mds-0.1/java-api/build/build.mk
@@ -85,7 +85,7 @@ $(project_name)_ignore_src_dirs ?= $(ign
 ignore_src_dir_patterns := $(addprefix %/,$($(project_name)_ignore_src_dirs))
 
 static_src_dirs := $(src_dir) $(filter-out $(ignore_src_dir_patterns),$(static_src_subdirs))
-generated_src_dirs := $(generated_src_dir) $(filter-out $(ignore_src_dir_patterns),$(generated_src_subdirs))
+generated_src_dirs := $(generated_src_dir) $(filter/out $(ignore_src_dir_patterns),$(generated_src_subdirs))
 
 src_dirs := $(static_src_dirs) $(generated_src_dirs)
 
@@ -193,7 +193,8 @@ std_flag = $(if $(call not_defined,$(pro
 cpp_defines ?=
 $(project_name)_cpp_defines ?= $(cpp_defines)
 
-extra_incl_dirs ?=
+# crazy hacks but I'm not digging through these build scripts anymore
+extra_incl_dirs = ../../../include/ ../../jni/generated-src/ ../../jni/src
 $(project_name)_extra_incl_dirs ?= $(extra_incl_dirs)
 
 cpp_includes ?= $(addprefix -I,$($(project_name)_incl_dirs) $($(project_name)_extra_incl_dirs))
@@ -220,7 +221,7 @@ $(project_name)_debug_flag ?= $(debug_fl
 extra_cxx_flags ?= -Wall -fmessage-length=0 -pthread -fPIC
 $(project_name)_extra_cxx_flags ?= $(extra_cxx_flags)
 
-CPPFLAGS ?= \
+CPPFLAGS += \
 	$(std_flag) \
 	$($(project_name)_cpp_defines) \
 	$($(project_name)_cpp_includes) \
@@ -228,7 +229,7 @@ CPPFLAGS ?= \
 	$($(project_name)_extra_cpp_flags)
 $(project_name)_CPPFLAGS ?= $(CPPFLAGS)
 
-CXXFLAGS ?= \
+CXXFLAGS += \
 	$($(project_name)_opt_flag) \
 	$($(project_name)_debug_flag) \
 	$($(project_name)_extra_cxx_flags)
@@ -396,7 +397,7 @@ $(obj_root_dir)/%.o: $(src_dir)/%.cpp
 	@echo Building file: $<
 	-@mkdir -p $(dir $@)
 	-@mkdir -p $(dir $(call obj_to_dep,$@))
-	$(CXX) -c $($(project_name)_CPPFLAGS) $($(project_name)_CXXFLAGS) -o "$@" "$<"
+	$(CXX) -c $($(project_name)_CPPFLAGS) $($(project_name)_CXXFLAGS) -I../../include -o "$@" "$<"
 	@echo Finished building: $<
 	@echo
 
Index: mds-0.1/java-api/build/project.mk
===================================================================
--- mds-0.1.orig/java-api/build/project.mk
+++ mds-0.1/java-api/build/project.mk
@@ -28,11 +28,11 @@ include ../defs.mk
 
 
 project_name = mds_java_api
-projects_used = mds_core mpgc ruts
+projects_used = mds_core
 
 mpgc_project_dir ?= $(if $(wildcard $(git_base_dir)/mpgc/build),$(git_base_dir)/mpgc,$(git_base_dir)/gc)
 ruts_project_dir ?= $(if $(wildcard $(mpgc_project_dir)/ruts/build),$(mpgc_project_dir)/ruts,$(git_base_dir)/common)
-mds_core_project_dir ?= $(if $(wildcard $(git_base_dir)/mds/build),$(git_base_dir)/mds,$(git_base_dir)/core)
+mds_core_project_dir = ../../../
 
 real_project_dir := $(project_dir)
 project_dir := $(real_project_dir)/jni
